### PR TITLE
feat: Added options  MaxMessageSize and Resource for step container for processing large messages

### LINKED ISSUE #177: Test with large messages (>1Mb)
| Source | |
|---|---|
| HTTP | | 
| Kafka | | 
| STAN | |

| Sink | |
|---|---|
| HTTP | | 
| Kafka | | 
| STAN | |

### PR DESCRIPTION
**Pull request by sarabala1979**
…

Signed-off-by: Saravanan Balasubramanian <sarabala1979@gmail.com>
closes #177

### CODE REVIEW

File: api/v1alpha1/cat.go [hunk 1]
```diff
@@ -4,11 +4,18 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-type Cat struct{}
+type Cat struct {
+	AbstractStep `json:",inline" protobuf:"bytes,1,opt,name=abstractStep"`
+}
 
 func (m Cat) getContainer(req getContainerReq) corev1.Container {
+	resources := standardResources
```
Comment (alexec) [inline api/v1alpha1/cat.go:12]: Do we need this, doesn’t the default populate this?

File: docs/BIG_PAYLOAD.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#Processing Big Payload >1MB
```
Comment (alexec) [inline docs/BIG_PAYLOAD.md:1]: This is not valid markdown, you need a space after the #.

File: runner/sidecar/sink/kafka/kafka.go [hunk 3]
```diff
@@ -48,6 +48,11 @@ func New(ctx context.Context, secretInterface corev1.SecretInterface, x dfv1.Kaf
 		return nil, err
 	}
 
+	if x.MaxMessageBytes > 0 {
+		config.Producer.MaxMessageBytes = int(x.MaxMessageBytes)
+	}
+	logger.Info("Kafka config Producer.MaxMessageBytes	", "value", config.Consumer.Fetch.Max)
```
Comment (alexec) [inline runner/sidecar/sink/kafka/kafka.go:54]: Do you want to trim the spaces?

File: t.yaml [hunk 4]
```diff
@@ -0,0 +1,171 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (alexec) [inline t.yaml:1]: Delete this file?

File: test/kafka-stress/kafka_stress_test.go [hunk 5]
```diff
@@ -5,6 +5,9 @@ package kafka_stress
 import (
 	"testing"
 
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
```
Comment (alexec) [inline test/kafka-stress/kafka_stress_test.go:9]: Minor - you could dot import this if you like.

**Review by alexec — APPROVED** [review]
Some minor comments.

File: api/v1alpha1/cat.go (hunk 1 above)
Comment (sarabala1979) [reply api/v1alpha1/cat.go:12]: no

**Pull request merged**